### DISCUSSION TITLE
'constant' and 'immutable' readouts

### DISCUSSION THREAD

**Discussion by CodeTivity08** [Cyfrin/foundry-full-course-cu#142] [Q&A] [answered]
Is this right?
```
    uint256 public constant MINIMUM_USD = 50 * 1e18;
    // 2451 gas - non-constant
    // 351 gas - constant

```
    address public immutable i_owner;
    // 444 gas - immutable
    // 2580 gas - non-immutable
```
I'm sure I'm doing it right, and it compiles and runs as shown, but are these gas costs right?

**Comment by rohit-bhalerao-7** [comment, marked as answer]
I believe the gas costs are right. But you can cross-check those in Remix if you want.

Please read this to understand it better: [https://medium.com/@ajith-m-doodlebug/difference-between-constant-and-immutable-in-solidity-b3a8adc4904f](url)

**Reply by CodeTivity08** [reply]
Every bit helps, thanks. The differences looked minimal when Patrick did it in the lessons (lesson 4, that is), so it looked unusual when I got these results.

**Reply by rohit-bhalerao-7** [reply]
Happy to help! Yeah right

**Reply by rohit-bhalerao-7** [reply]
Hey, I got to know one more thing `constant` and `immutable `are part of the `bytecode` of the contract itself which gets executed at the machine level, its like a pointer pointing to its value.  They are not stored in storage.

**Reply by CodeTivity08** [reply]
I know. I just completed lesson 4, but I think I need to learn Javascript alongside this course. It's gonna be a bit of a waste to learn all this Smart Contract goodness but have no knowledge of building apps (dApps, in this case) to utilize Blockchain's full potential.

**Reply by rohit-bhalerao-7** [reply]
Agreed. It's better to know JS, even I need to learn it, I started the Hardhat course which was in JS but couldn't catch the concepts due lack of knowledge in JS so couldn't complete the course.

**Reply by CodeTivity08** [reply]
Goodluck. Hope to congratulate each other at the end.
